apps/examples/testcase/le_tc/compression/tc_lzma.c

```diff
@@ -0,0 +1,138 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <crc32.h>
+#include <tinyara/lzma/LzmaLib.h>
+
+unsigned int propsSize = LZMA_PROPS_SIZE;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: tc_2Kb
+ *
+ * Description:
+ *   Check and verify the checksum of a file of size 2KB after compression and decompression
+ *
+ * Returned Value:
+ *   None
+ ****************************************************************************/
+void tc_2Kb(void)
+{
+
+	int i, ret;
+	unsigned int ini_t, after_t;
+	unsigned int writesize = 2048;
+	unsigned int max = 2048;
+	unsigned char arr[2048];
+	unsigned char out_buf[2048 + LZMA_PROPS_SIZE];
+
+	for (i = 0; i < 2048; i++) {
+		arr[i] = 'A' + ((rand()) % 26);
+	}
+
+	ini_t = crc32(arr, 2048);
+
+	ret = LzmaCompress(&out_buf[LZMA_PROPS_SIZE], &writesize, arr, 2048, out_buf, &propsSize, 0, 1 << 13, -1, -1, -1, -1, 1);
+
+	if (ret != SZ_OK) {
+		printf("LZMA Compress failed, ret = %d\n", ret);
```

Comment:
All tc should follow the TC_ASSERT_XXX or TC_SUCCESS_RESULT macros.

Reply:
Done